stubs/python-jenkins/jenkins/__init__.pyi

```diff
@@ -0,0 +1,216 @@
+    def jenkins_open_stream(self, req: Request, add_crumb: bool = True, resolve_auth: bool = True) -> Response: ...
+    def jenkins_request(
+        self, req: Request, add_crumb: bool = True, resolve_auth: bool = True, stream: bool | None = None
+    ) -> Response: ...
+    def get_queue_item(self, number: int, depth: int = 0) -> Mapping[Any, Any]: ...
+    def get_build_info(self, name: str, number: int, depth: int = 0) -> Mapping[Any, Any]: ...
+    def get_build_env_vars(self, name: str, number: int, depth: int = 0) -> Mapping[Any, Any] | None: ...
+    def get_build_test_report(self, name: str, number: int, depth: int = 0) -> Mapping[Any, Any] | None: ...
+    def get_build_artifact(self, name: str, number: int, artifact: str) -> Mapping[Any, Any]: ...
+    def get_build_artifact_as_bytes(self, name: str, number: int, artifact: str) -> bytes: ...
+    def get_build_stages(self, name: str, number: int) -> Mapping[Any, Any]: ...
+    def get_queue_info(self) -> Mapping[Any, Any]: ...
+    def cancel_queue(self, id: int) -> None: ...
+    def get_info(self, item: str = "", query: str | None = None) -> Mapping[Any, Any]: ...
+    def get_whoami(self, depth: int = 0) -> Mapping[Any, Any]: ...
+    def get_version(self) -> str: ...
+    def get_plugins_info(self, depth: int = 2) -> Mapping[Any, Any]: ...
+    def get_plugin_info(self, name: str, depth: int = 2) -> Mapping[Any, Any]: ...
+    def get_plugins(self, depth: int = 2) -> Mapping[Any, Any]: ...
+    def get_jobs(
+        self, folder_depth: int = 0, folder_depth_per_request: int = 10, view_name: str | None = None
+    ) -> Sequence[Mapping[str, str]]: ...
+    def get_all_jobs(
+        self, folder_depth: int | None = None, folder_depth_per_request: int = 10
+    ) -> Sequence[Mapping[str, str]]: ...
+    def copy_job(self, from_name: str, to_name: str) -> None: ...
+    def rename_job(self, from_name: str, to_name: str) -> None: ...
+    def delete_job(self, name: str) -> None: ...
+    def enable_job(self, name: str) -> None: ...
+    def disable_job(self, name: str) -> None: ...
+    def set_next_build_number(self, name: str, number: int) -> None: ...
+    def job_exists(self, name: str) -> bool: ...
+    def jobs_count(self) -> int: ...
+    def assert_job_exists(self, name: str, exception_message: str = "job[%s] does not exist") -> None: ...
+    def create_folder(self, folder_name: str, ignore_failures: bool = False) -> None: ...
+    def upsert_job(self, name: str, config_xml: str) -> None: ...
+    def check_jenkinsfile_syntax(self, jenkinsfile: str) -> Sequence[str]: ...
+    def create_job(self, name: str, config_xml: str) -> None: ...
+    def get_job_config(self, name: str) -> str: ...
+    def reconfig_job(self, name: str, config_xml: str) -> None: ...
+    def build_job_url(
+        self, name: str, parameters: Mapping[str, Any] | Sequence[tuple[str, Any]] | None = None, token: str | None = None
+    ) -> str: ...
```

Comment:
If `token` is truthy, then parameters can only be a `list` or `dict` as it's explicitly checked for.

```suggestion
    def build_job_url(
        self, name: str, parameters: dict[str, Any] | list[tuple[str, Any]] | None = None, token: str | None = None
    ) -> str: ...
```

If you still want to allow everything allowed by `urlencode` when `token` is falsy, you can consider using overloads:
```suggestion
	@overload
    def build_job_url(
        self, name: str, parameters: Mapping[str, Any] | Sequence[tuple[str, Any]] | None = None, token: Literal[''] | None = None
    ) -> str: ...
	@overload
    def build_job_url(
        self, name: str, parameters: dict[str, Any] | list[tuple[str, Any]] | None = None, token: str
    ) -> str: ...
```